File: runtimes/org.eclipse.edt.runtime.javascript/runtime/egl/lang/InvalidIndexException.js
/*******************************************************************************
 * Copyright © 2008, 2011 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * IBM Corporation - initial API and implementation
 *
 *******************************************************************************/

egl.createInvalidIndexException = function( /*string*/ messageID, /*string or array*/ inserts )
{
	if (typeof(inserts) != "string") {
		inserts = egl.getRuntimeMessage( messageID, inserts );
	}
	egl.exceptionThrown = true;
	var args = new Array();
	args.push( [ "messageID", messageID || "" ] );
	args.push( [ "message", inserts || "" ] );
	args.push( [ "index", arguments[ 2 ] || 0 ] );
	return new egl.egl.lang.InvalidIndexException( args );
}

egl.defineClass( "egl.lang", "InvalidIndexException", "egl.lang", "AnyException", {
	
	//InvalidIndexException requires the following additional data:
	//index
	
	"eze$$getChildVariables" : function() {
		var childVars = this.eze$$superClass.prototype.eze$$getChildVariables.call( this );
		childVars.push({name : "index", value : this.index, type : "eglx.lang.EInt"});
		return childVars;
	}
	
});

File: runtimes/org.eclipse.edt.runtime.java/src/eglx/lang/InvalidIndexException.java
/*******************************************************************************
 * Copyright © 2011, 2013 IBM Corporation and others.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 * IBM Corporation - initial API and implementation
 *
 *******************************************************************************/
package eglx.lang;
import org.eclipse.edt.javart.resources.*;
import org.eclipse.edt.javart.*;
import eglx.lang.AnyException;
import org.eclipse.edt.runtime.java.eglx.lang.EInt;
import java.lang.Integer;
@SuppressWarnings("unused")
@javax.xml.bind.annotation.XmlRootElement(name="InvalidIndexException")
public class InvalidIndexException extends eglx.lang.AnyException {
	private static final long serialVersionUID = 10L;
	@javax.xml.bind.annotation.XmlTransient
	public int index;
	public InvalidIndexException() {
		super();
	}
	{
		ezeInitialize();
	}
	public void ezeCopy(Object source) {
		ezeCopy((InvalidIndexException) source);
	}
	public void ezeCopy(eglx.lang.AnyValue source) {
		this.index = ((InvalidIndexException) source).index;
	}
	public InvalidIndexException ezeNewValue(Object... args) {
		return new InvalidIndexException();
	}
	public void ezeSetEmpty() {
		index = 0;
	}
	public boolean isVariableDataLength() {
		return false;
	}
	public void loadFromBuffer(ByteStorage buffer, Program program) {
	}
	public int sizeInBytes() {
		return 0;
	}
	public void storeInBuffer(ByteStorage buffer) {
	}
	public void ezeInitialize() {
		index = 0;
	}
	@org.eclipse.edt.javart.json.Json(name="index", clazz=EInt.class, asOptions={})
	public int getIndex() {
		return index;
	}
	public void setIndex(int ezeValue) {
		index = ezeValue;
	}
}
